### PR TITLE
(RE-7587) (RE-7593) Windows fixes to mco and pxp-agent

### PR DESCRIPTION
**Pull request by McdonaldSeanp**
~~This PR requires https://github.com/puppetlabs/pxp-agent/pull/448~~

### CODE REVIEW

**Comment by McdonaldSeanp** [comment]
pxp-agent PR was merged, this is ready

**Comment by McdonaldSeanp** [comment]
matrix build at http://jenkins-release.delivery.puppetlabs.net/job/vanagon_generic_job/539/

File: configs/components/marionette-collective.rb [hunk 1]
```diff
@@ -81,14 +81,15 @@
...
 
   flags = " --bindir=#{settings[:bindir]} \
+            --sbindir=#{settings[:bindir]} \
             --sitelibdir=#{settings[:ruby_vendordir]} \
             --ruby=#{File.join(settings[:bindir], 'ruby')} "
 
   if platform.is_windows?
-    pkg.install_file "ext/windows/daemon.bat", "#{settings[:bindir]}/mco_daemon.bat"
-    pkg.add_source("file://resources/files/windows/mco.bat", sum: "2d29af9c926dcf8b50ae9ac1bdb18e1f")
+    pkg.add_source("file://resources/files/windows/mco.bat")
```
Comment (melissa) [inline configs/components/marionette-collective.rb]: `add_source` no longer needs a checksum?

Comment (McdonaldSeanp) [inline configs/components/marionette-collective.rb]: Yep, it no longer requires checksums for local files

File: configs/components/ruby.rb [hunk 2]
```diff
@@ -189,6 +189,21 @@
...
+    # (puppet.bat ,mco.bat etcc) the location of openssl away from where ruby is
+    # installed will cause a failure. Specifically this is meant to help services like
+    # mco that require openssl but don't have access to environment.bat. Refer to
+    # https://tickets.puppetlabs.com/browse/RE-7593 for details on why this causes
+    # failures and why these copies fix that.
+    #                   -Sean P. McDonald 07/01/2016
+    pkg.install do
+      [
+        "cp #{settings[:prefix]}/bin/ssleay32.dll #{settings[:ruby_bindir]}",
+        "cp #{settings[:prefix]}/bin/libeay32.dll #{settings[:ruby_bindir]}",
```
Comment (melissa) [inline configs/components/ruby.rb]: Hmm, I'm wondering if this change makes more sense to do in the openssl.rb file? Or is it that we need to make these files available specifically in the ruby bindir? I'm not sure what makes more sense...

Comment (melissa) [inline configs/components/ruby.rb]: So, I guess I'm wondering is: should we default to install openssl where ruby is installed?

File: configs/components/marionette-collective.rb (hunk 1 above)
Comment (melissa) [inline configs/components/marionette-collective.rb]: solid!

**Comment by melissa** [comment]
👍  aside from question on the default openssl install location. But I think this is a very reasonable change

File: configs/components/ruby.rb (hunk 2 above)
Comment (McdonaldSeanp) [inline configs/components/ruby.rb]: I believe that will break the puppet service in the same way. From what I know, I believe they need to exists in both places ... I will check it out.

Comment (melissa) [inline configs/components/ruby.rb]: If that breaks things, then don't worry about it

**Pull request merged**